### PR TITLE
allow multiple MySQL endpoints

### PR DESCRIPTION
**Pull request by eryshev**
We allow to configure multiple MySQL endpoints for failover when one of them is unavailable.
When starting a new connection, the driver always tries to connect to the primary host first and, if required, fails over to the secondary hosts on the list sequentially when communication problems are experienced.
https://dev.mysql.com/doc/connector-j/5.1/en/connector-j-config-failover.html

### CODE REVIEW

**Comment by dufrannea** [comment]
I feel it is not really cuttle's concern if you have HA or failover, and thus not its responsibility to build the connectionString. Maybe we should allow to specify a connectionString directly rather than db name / host / port ?

File: core/src/main/scala/com/criteo/cuttle/Database.scala [hunk 1]
```diff
@@ -16,33 +16,42 @@ import java.util.concurrent.{Executors, TimeUnit}
...
+/** Configuration of MySQL endpoint.
   *
   * @param host JDBC driver host
   * @param port JDBC driver port
+  */
+case class DBLocation(host: String, port: Int)
+
+/** Configuration for the MySQL database used by Cuttle.
+  *
   * @param database JDBC database
```
Comment (vguerci) [inline core/src/main/scala/com/criteo/cuttle/Database.scala]: locations doc

File: core/src/main/scala/com/criteo/cuttle/Database.scala [hunk 2]
```diff
@@ -178,24 +187,30 @@ private[cuttle] object Database {
     } yield ())
 
   private val connections = collection.concurrent.TrieMap.empty[DatabaseConfig, XA]
-  def connect(c: DatabaseConfig): XA =
+  def connect(dBConfig: DatabaseConfig): XA = {
```
Comment (vguerci) [inline core/src/main/scala/com/criteo/cuttle/Database.scala]: OPT: case, should be `dbConfig`

File: devloop.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ let database = runServer({
   name: 'mysqld',
   httpPort: 3388,
   sh: 'java -cp `cat /tmp/classpath_com.criteo.cuttle.localdb` com.criteo.cuttle.localdb.LocalDB',
-}).dependsOn(generateClasspath)
+}).dependsOn(generateClasspath);
```
Comment (vguerci) [inline devloop.js]: OPT: is that `;` really necessary?

Comment (dufrannea) [reply devloop.js]: Yes it is ! :)

**Pull request merged**